File: HardwareInfoCollector/models/ssd.py
from models.hardware import *


class SSD(Hardware):
    Capacity: int
    ReadSpeed: int
    WriteSpeed: int

    def __init__(self, product_type: int, model: str, price: int, link: str,
                 capacity: int, read_speed: int, write_speed: int):
        super().__init__(product_type, model, price, link)
        self.Capacity = capacity
        self.ReadSpeed = read_speed
        self.WriteSpeed = write_speed

    def __str__(self):
        return f'{self.ProductType} {self.Model} {self.Price} {self.Link} {self.Capacity} ' \
               f'{self.ReadSpeed} {self.WriteSpeed}'

File: BuildGenerator/src/Bll/Models/Hardware/Ssd.cs
namespace Bll.Models.Hardware;

public class Ssd : Abstractions.Hardware
{
    public double Price { get; set; }
    public string Link { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;

    public int Capacity { get; set; }
    public int ReadSpeed { get; set; }
    public int WriteSpeed { get; set; }

    public Ssd()
    {
        
    }
    
    public Ssd(double price, string link, string model, int capacity, int readSpeed, int writeSpeed)
    {
        Price = price;
        Link = link;
        Model = model;
        Capacity = capacity;
        ReadSpeed = readSpeed;
        WriteSpeed = writeSpeed;
    }
}